csrc/scheduler/tools/loop_domain_scheduler.cpp

```diff
@@ -460,23 +539,21 @@ void scheduleLoopDomainsBy(
       continue;
     }
 
-    const auto& existing_ids =
-        replay_dir_tv == Direction::Forward ? input_ids : output_ids;
-
-    // Clone inputs or outputs
-    auto& new_ids =
-        replay_dir_tv == Direction::Forward ? output_ids : input_ids;
-    const auto& ref_of_ids_to_generate = replay_dir_tv == Direction::Forward
-        ? transform->outputs()
-        : transform->inputs();
+    // When the direction is forward, the TensorView transform
+    // APIs, e.g., TensorView::split, can be used, which doesn't need
+    // to use TensorView::setLoopDomain.
+    if (replay_dir_tv == Direction::Forward) {
+      ReplayForwardTransformOnLoopDomain::replayAs(tv, input_ids, transform);
+      continue;
+    }
 
-    for (const auto& ref_id : ref_of_ids_to_generate) {
+    NVF_ERROR(input_ids.empty());
```

Comment:
Below changes are just simplification as `replay_dir_tv` can only be `Direction::Backward`